cocos/deprecated/CCDeprecated.h

```diff
@@ -32,8 +32,59 @@
 #define __COCOS2D_CCDEPRECATED_H__
 
 #include <math.h>
-#include "math/CCGeometry.h"
+#include "deprecated/CCSet.h"
```

Comment:
Oops, It can't be done.
`CCDeprecated.h` needs lots of headers to be included.
And it uses the depended classes directly in `CCDeprecated.h`.

I think it is harmless to add required headers in `CCDeprecated.h`.
So when user wants to enable deprecated features, just include `CCDeprecated.h`, rather than, as `CCDeprecated.h` was required, include the whole `cocos2d.h`.

Include of `cocos2d.h` will cause lots of rebuild when touched any of cocos headers.